### PR TITLE
Implement LAB 4 Feature Numeronium

### PR DESCRIPTION
**Pull request by jatheesh**
Fixes #67, fixes #70  fixes #71 

#### Changes proposed in this PR:
LAB 4  and LAB 5 
- Add numeroniyum Feature and associated test cases
- Add Stringer Feature and associated test cases

### CODE REVIEW

File: 04_numeronym/Jat/numeroniyum.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, numeronyms("", "ab", "abc", "amala", "Kubernetes"))
```
Comment (juliaogris) [inline 04_numeronym/Jat/numeroniyum.go]: Not the requested input.

File: 04_numeronym/Jat/numeroniyum.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+func numeronyms(vals ...string) []string {
```
Comment (juliaogris) [inline 04_numeronym/Jat/numeroniyum.go:16]: Nice! 👍

File: 04_numeronym/Jat/numeroniyum_test.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	// Given
+	r := require.New(t)
+
+	// When
+	m := numeronyms()
+
+	// Then
+	expected := 0
+	actual := len(m)
+	r.Equalf(expected, actual, "Unexpected output in main()")
```
Comment (juliaogris) [inline 04_numeronym/Jat/numeroniyum_test.go:55]: Bad failure message. 
Try setting up using [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests) and have as one of the "row" values the failure message - that should simplify/shorten your code a lot.

Also make sure your code works with unicode, e.g. emojis

**Review by juliaogris — COMMENTED** [review]
Just a couple of minor things.

File: 04_numeronym/Jat/numeroniyum_test.go (hunk 3 above)
Comment (juliaogris) [reply 04_numeronym/Jat/numeroniyum_test.go:55]: Add test cases for unicode at beginning and end and white space at beginning and end please

**Review by juliaogris — CHANGES_REQUESTED** [review]

**Comment by juliaogris** [comment]
Hi,
There has been  no activity in the last 20 days so I will close this PR.
Feel free to open it again if you have something to add.
Thanks!
Julia
